### PR TITLE
Bluntsord/realtime notes updated

### LINKED ISSUE #218: [PE-D] Notes not saved if User does not execute another command
If I add some notes and quit the app before typing any other command, my notes will not be saved. I.e. DonnaFin only saves after every command execution

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: yongxiangng/ped#16

### PR DESCRIPTION
**Pull request by Bluntsord**
Make notes have realtime update

Fixes #218 

Previously we did not save everytime we wrote the notes. We were lucky that it was caught when running commands in logicManager execute. Hence, it saving only when we ran a new command.

image

Therefore just add the saveAddressbook command to one of the functions called by personadapter

### CODE REVIEW

**Comment by bharathcs** [comment]
Niceeeee. This makes sense. But tbh we shouldn't swallow an error from `saveAddressBook`. 

If the file gets corrupted / we have some IO problem, it should throw an exception and shut down the app. 

This can either be just an uncaught runtime exception, or even better, call on the method fatalerror and shutdown which is either in `UiManager` or `MainWindow`.

**Review by bharathcs — CHANGES_REQUESTED** [review]
Niceeeee. This makes sense. But tbh we shouldn't swallow an error from `saveAddressBook`. 

If the file gets corrupted / we have some IO problem, it should throw an exception and shut down the app. 

This can either be just an uncaught runtime exception, or even better, call on the method fatalerror and shutdown which is either in `UiManager` or `MainWindow`.

**Comment by Bluntsord** [comment]
> Niceeeee. This makes sense. But tbh we shouldn't swallow an error from `saveAddressBook`.
> 
> If the file gets corrupted / we have some IO problem, it should throw an exception and shut down the app.
> 
> This can either be just an uncaught runtime exception, or even better, call on the method fatalerror and shutdown which is either in `UiManager` or `MainWindow`.

So I should make the function throw an error? Can cfm?

**Comment by bharathcs** [comment]
> So I should make the function throw an error? Can cfm?

You can have the PersonAdapter's edit throw a runtime exception (unchecked). This would cause the app to shut down and the command prompt terminal to show an exception error.

Or you can replace printstacktrace that you wrote here with `((UiManager) ui).showFatalErrorDialogAndShutdown()` 

The first way is the best way but idrc, the first way is easier to write and doesn't require any changes. We can do the first way to save time now since we're so close to the end.

File: src/main/java/donnafin/model/ModelManager.java [hunk 1]
```diff
@@ -118,10 +118,13 @@ public void addPerson(Person person) {
     }
 
     @Override
-    public void setPerson(Person target, Person editedPerson) {
+    public void setPerson(Person target, Person editedPerson) throws IOException {
         CollectionUtil.requireAllNonNull(target, editedPerson);
 
         addressBook.setPerson(target, editedPerson);
+
+        //Need to run save addressBook if not it wont save.
```
Comment (bharathcs) [inline src/main/java/donnafin/model/ModelManager.java:126]:
```suggestion
        // Required to save changes from the model representation 
        // in Java memory to the user's local storage.
```

**Review by bharathcs — APPROVED** [review]

**Review by bharathcs — APPROVED** [review]
Nice work!!

File: src/main/java/donnafin/logic/PersonAdapter.java [hunk 2]
```diff
@@ -71,7 +72,12 @@ public void edit(Name newName) {
                 curr.getAssets()
         );
         this.subject = personToEdit;
-        model.setPerson(curr, personToEdit);
+        try {
+            model.setPerson(curr, personToEdit);
+            model.saveAddressBook();
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage());
+        }
```
Comment (bharathcs) [inline src/main/java/donnafin/logic/PersonAdapter.java:80]:
```suggestion
        trySavingAddressBook();
```

:warning: Do not commit this directly, this is not complete.

What if we replace all these duplicated code fragments with a call to a method that does exactly this. A private method `trySavingAddressBook()` in `PersonAdapter` would work right?

**Review by bharathcs — APPROVED** [review]
Nice work, feel free to commit if you like but I have a suggestion to reduce code duplication if you like.

**Review by bharathcs — APPROVED** [review]
LGTM!

**Pull request merged**